assignment4/solvers/policy_iteration.py

```diff
@@ -52,8 +52,8 @@ def step(self):
             self._policy[s] = np.eye(self._env.nA)[best_a]
 
         # If we've gone through a few steps but have not improved, consider us converged
-        if delta > self._last_delta and self._steps > 10:
-            self._policy_stable = True
+        if self._steps < 10 and self._policy_stable:
```

Comment:
This seemed more correct to me than using the delta variable. First, it looks closer to what was done in the https://github.com/dennybritz/reinforcement-learning/blob/master/DP/Policy%20Iteration%20Solution.ipynb implementation that this is based on. Second, "delta" is reset on every loop through the states, so this is really only checking the last delta in the loop.

Reply:
Maybe "delta" isn't needed at all with this change? Or maybe the change should be such that delta accumulates on each iteration through the policy instead of being reset?

Reply:
see Sutton/Barto page 80: they are using a threshold value for convergence